app/models/order_item.rb

```diff
@@ -16,7 +16,9 @@ class OrderItem < ApplicationRecord
   belongs_to :portfolio_item
   has_many :progress_messages, :dependent => :destroy
   has_many :approval_requests, :dependent => :destroy
+
   before_create :set_defaults
+  after_create :insights_request_progress_message
```

Comment:
Should this be moved out of `OrderItem` and potentially be added as an operation in a service? Maybe`submit_order_item`?
The ProgressMessage entry isn't needed for the `OrderItem`which makes the callback a potential complication for a third party 'logging' operation.

Reply:
@syncrou The SubmitOrder class would be too late in the process.  Since 3Scale adds the `insights_request_id` on the API call that adds the item to the order we should be able to use it for actions before we call Topology to launch the Job.  For example, collecting tags from Topo/Sources and running the Approval workflows.

The idea was to provide this ID as early as possible to track any service-to-service activity that happens on behalf of this transaction.

If it does not belong in the object then I think the `AddToOrder` service class is the next best choice.  Thoughts?

Reply:
`AddToOrder` sounds fine to me.